com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -129,24 +132,59 @@ internal unsafe void __sendClientRpc(FastBufferWriter writer, uint rpcMethodId,
             };
             int messageSize;
 
+            // We check to see if we need to shortcut for the case where we are the host/server and we can send a clientRPC
+            // to ourself. Sadly we have to figure that out from the list of clientIds :( 
+            ulong serverClientId = ulong.MaxValue;
+
             if (rpcParams.Send.TargetClientIds != null)
             {
                 // Copy into a localArray because SendMessage doesn't take IEnumerable, only IReadOnlyList
                 ulong* localArray = stackalloc ulong[rpcParams.Send.TargetClientIds.Count()];
                 var index = 0;
                 foreach (var clientId in rpcParams.Send.TargetClientIds)
                 {
+                    if (clientId == NetworkManager.ServerClientId)
+                    {
+                        serverClientId = clientId;
+                        continue;
+                    }
+
                     localArray[index++] = clientId;
                 }
+
                 messageSize = NetworkManager.SendMessage(message, networkDelivery, localArray, index, true);
```

Comment:
I see the `serverCanSendToServerId` is true here which prevents a second walk of the list, but on first read it's counter-intuitive to remove the server from the list and then say `serverCanSendToServerId = true` here. Maybe a comment?

Reply:
I think `serverCanSendToServerId` param should be removed from `SendMessage`. It was added specifically and only for RPCs and is now no longer relevant.